src/components/auth/register.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react'
+import { navigate } from 'gatsby'
+import { createUserWithEmailAndPassword, updateProfile } from 'firebase/auth'
+import {
+    RegisterForm,
+    AuthInput,
+    LoginButton,
+    RegisterButton,
+    ResetButton,
+    ControlArea,
+} from './styles'
+import { auth } from './firebase'
+import { localize } from 'components/localization'
+import { Header } from 'components/elements'
+
+type TAuthRegister = {
+    setStatus: (status: string) => void
+}
+
+const userInit = {
+    email: '',
+    password: '',
+    confirmPassword: '',
+    displayName: '',
+}
+
+const AuthRegister = ({ setStatus }: TAuthRegister) => {
+    const [user, setUser] = useState(userInit)
+    const [error, setError] = useState('')
+
+    const validatePassword = () => {
+        let isValid = true
+        if (user.password !== '' && user.confirmPassword !== '') {
+            if (user.password !== user.confirmPassword) {
+                isValid = false
+                setError('Passwords does not match')
+            }
+        }
+        return isValid
+    }
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+
+        try {
+            setError('')
+            if (validatePassword()) {
+                await createUserWithEmailAndPassword(auth, user.email, user.password)
+            }
+        } catch (err) {
+            setError(err.message)
+        }
+        await updateProfile(auth.currentUser, { displayName: user.displayName })
+        navigate(`/`)
+        setStatus('')
+        setUser(userInit)
+    }
+
+    return (
+        <>
+            <div>
+                {error && <div>{error}</div>}
+                <form onSubmit={handleSubmit}>
+                    <RegisterForm>
+                        <AuthInput
+                            type="text"
+                            name="userName"
+                            value={user.displayName}
+                            required
+                            onChange={(e) => setUser({ ...user, displayName: e.target.value })}
+                            placeholder="Username"
+                        />
+                        <AuthInput
+                            type="email"
+                            name="email"
+                            value={user.email}
+                            required
+                            onChange={(e) => setUser({ ...user, email: e.target.value })}
+                            placeholder="E-mail Address"
```

Comment:
is E-mail hyphen needed between in the placeholder?

Reply:
We have no requirements for that, that's why we can rename it 👍